workshops/dlt/03_dlt_workshop_streaming.py

```diff
@@ -0,0 +1,22 @@
+import requests
+import json
+
+def download_and_yield_rows(url):
+    response = requests.get(url, stream=True)
+    response.raise_for_status()  # Raise an HTTPError for bad responses
+
+    for line in response.iter_lines():
+        if line:
+            yield json.loads(line)
+
+# Replace the URL with your actual URL
+url = "https://storage.googleapis.com/dtc_zoomcamp_api/yellow_tripdata_2009-06.jsonl"
+
+i = 0
+# Use the generator to iterate over rows with minimal memory usage
+for row in download_and_yield_rows(url):
+    #process each row as needed
+    print(row)
+    i += 1
+    if i == 10:
+        break
```

Comment:
this can be `while` I think

```python
i = 0

while i != 0:
   # do something
   i+=1
```

Reply:
Fix : Reslove 03_dlt_workshop_streaming.py file use while loop.

Reply:
![Screenshot 2024-02-12 154936](https://github.com/omjirapat/data-engineering-zoomcamp/assets/156293220/dafe890a-1f7c-40a4-b790-955b2ce047ad)